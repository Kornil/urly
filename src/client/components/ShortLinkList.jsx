import React from "react";
import PropTypes from "prop-types";
import styled from "react-emotion";

const StyledShortLinkList = styled.div`
  margin: 40px 0;
`;

const ShortLinkList = ({ cachedUrls }) => (
  <StyledShortLinkList>
    {Object.keys(cachedUrls).map(key => (
      <p key={key}>
        <a href={cachedUrls[key].url} rel="noopener noreferrer" target="_blank">
          {key} - {cachedUrls[key].url}
        </a>
      </p>
    ))}
  </StyledShortLinkList>
);

ShortLinkList.propTypes = {
  cachedUrls: PropTypes.objectOf(PropTypes.shape({ url: PropTypes.string }))
    .isRequired
};

export default ShortLinkList;
