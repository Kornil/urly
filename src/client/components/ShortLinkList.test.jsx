import { shallow } from "enzyme";
import React from "react";

import ShortLinkList from "./ShortLinkList";

it("it renders", () => {
  const wrapper = shallow(
    <ShortLinkList
      cachedUrls={{
        MDL7g1YiM: {
          url: "http://google.com"
        }
      }}
    />
  );
  expect(wrapper).toMatchSnapshot();
});
