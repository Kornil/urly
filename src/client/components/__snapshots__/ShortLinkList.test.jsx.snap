// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ShortLinkList
    cachedUrls={
      Object {
        "MDL7g1YiM": Object {
          "url": "http://google.com",
        },
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <p>
          <a
            href="http://google.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            MDL7g1YiM
             - 
            http://google.com
          </a>
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <a
            href="http://google.com"
            rel="noopener noreferrer"
            target="_blank"
          >
            MDL7g1YiM
             - 
            http://google.com
          </a>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "MDL7g1YiM",
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "MDL7g1YiM",
              " - ",
              "http://google.com",
            ],
            "href": "http://google.com",
            "rel": "noopener noreferrer",
            "target": "_blank",
          },
          "ref": null,
          "rendered": Array [
            "MDL7g1YiM",
            " - ",
            "http://google.com",
          ],
          "type": "a",
        },
        "type": "p",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <p>
            <a
              href="http://google.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              MDL7g1YiM
               - 
              http://google.com
            </a>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <a
              href="http://google.com"
              rel="noopener noreferrer"
              target="_blank"
            >
              MDL7g1YiM
               - 
              http://google.com
            </a>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "MDL7g1YiM",
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "MDL7g1YiM",
                " - ",
                "http://google.com",
              ],
              "href": "http://google.com",
              "rel": "noopener noreferrer",
              "target": "_blank",
            },
            "ref": null,
            "rendered": Array [
              "MDL7g1YiM",
              " - ",
              "http://google.com",
            ],
            "type": "a",
          },
          "type": "p",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
